concent_api/api-integration-multi-threads-test.py

```diff
@@ -0,0 +1,201 @@
+from api_testing_common import count_fails
+from api_testing_common import PROVIDER_PRIVATE_KEY
+from api_testing_common import REQUESTOR_PRIVATE_KEY
+from api_testing_common import api_request
+from api_testing_common import create_signed_task_to_compute
+from api_testing_common import run_tests
+from api_testing_common import timestamp_to_isoformat
+from common.helpers import get_current_utc_timestamp
+from common.helpers import sign_message
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "concent_api.settings")
+
+REPORT_COMPUTED_TASK_SIZE = 10
+
+NUMBER_OF_TESTING_THREADS = 3  # changing this variable remember to correct number of expected responses
+MAXIMUM_WAITING_TIME_FOR_ALL_RESPONSES = 3  # seconds
+
+responses_global = []  # type: list
+
+
+def clear_responses():
+    global responses_global
+    responses_global.clear()
+
+
+def call_function_in_treads(
+    func,
+    number_of_threads: int,
+    cluster_url: str,
+    golem_message: Message,
+) -> None:
+    for i in range(number_of_threads):
+        thread = Thread(target=func, args=(cluster_url, golem_message,))
+        thread.start()
+
+
+def get_ack_report_computed_task(report_computed_task: ReportComputedTask) -> AckReportComputedTask:
+    return AckReportComputedTask(report_computed_task=report_computed_task)
+
+
+def get_force_get_task_result(report_computed_task: ReportComputedTask) -> ForceGetTaskResult:
+    return message.concents.ForceGetTaskResult(report_computed_task=report_computed_task)
+
+
+def get_force_report_computed_task(report_computed_task: ReportComputedTask) -> ForceReportComputedTask:
+    return ForceReportComputedTask(report_computed_task = report_computed_task)
+
+
+def create_report_computed_task(
+    task_id: str,
+    subtask_id: str
+) -> ReportComputedTask:
+    current_time = get_current_utc_timestamp()
+
+    task_to_compute = create_signed_task_to_compute(
+        timestamp=timestamp_to_isoformat(current_time),
+        task_id=task_id,
+        subtask_id=subtask_id,
+        deadline=(current_time + 100),
+        price=10000,
+    )
+
+    report_computed_task = ReportComputedTaskFactory(
+        task_to_compute=task_to_compute,
+    )
+    sign_message(report_computed_task, PROVIDER_PRIVATE_KEY)
+    return report_computed_task
+
+
+def send_correct_force_report_computed_task(
+    cluster_url: str,
+    report_computed_task: ReportComputedTask,
+    expected_status: Optional[int] = None
+) -> None:
+    response = api_request(
+        cluster_url,
+        'send',
+        PROVIDER_PRIVATE_KEY,
+        CONCENT_PUBLIC_KEY,
+        get_force_report_computed_task(
+            report_computed_task=report_computed_task,
+        ),
+        headers={
+            'Content-Type': 'application/octet-stream',
+        },
+        expected_status=expected_status,
+    )
+    responses_global.append(response['error_code'] if isinstance(response, dict) else response)
+
+
+def send_correct_ack_report_computed_task(
```

Comment:
why not adding response to responses list in this case?

Reply:
this function is only to change the state before 'real' test. It would be not used in this place.